watcher/lib/kubernetes.go

```diff
@@ -34,27 +35,37 @@ func NewKubernetesClient() (*KubernetesClient, error) {
 }
 
 // WatchConfigMaps watches Kubernetes API for configmaps with the specified name prefix in the system
-// namespace and submits them to the returned channel
-func (c *KubernetesClient) WatchConfigMaps(ctx context.Context, prefix string) (chan string, error) {
-	watcher, err := c.ConfigMaps("kube-system").Watch(api.ListOptions{})
-	if err != nil {
-		return nil, trace.Wrap(err)
+// namespace and submits them to the provided channel
+func (c *KubernetesClient) WatchConfigMaps(ctx context.Context, prefix string, ch chan<- string) {
+	for {
+		select {
+		case <-time.After(time.Second):
```

Comment:
Why so often (If the restartWatch exits within a second - this will create quite a busy loop)? Is there a way to learn that the watch has expired?